kernel/src/transaction.rs

```diff
@@ -138,19 +135,21 @@ impl Transaction {
             .map(|txn| txn.into_engine_data(get_log_txn_schema().clone(), engine));
 
         // step one: construct the iterator of commit info + file actions we want to commit
-        let engine_commit_info = self
-            .commit_info
-            .as_ref()
-            .ok_or_else(|| Error::MissingCommitInfo)?;
-        let commit_info_actions = generate_commit_info(
-            engine,
-            self.operation.as_deref(),
-            self.commit_timestamp,
-            engine_commit_info.as_ref(),
+        let mut commit_info = CommitInfo::new(self.commit_timestamp);
+        commit_info.operation = Some(
+            self.operation
+                .clone()
+                .unwrap_or(UNKNOWN_OPERATION.to_string()),
         );
+        commit_info.kernel_version = Some(format!("v{KERNEL_VERSION}"));
+        commit_info.engine_info = self.engine_info.clone();
```

Comment:
feels like we may benefit from just a bigger constructor? pass in timestamp and optional engine info and optional operation? and we can do the `unwrap_or(UNKNOWN_OPERATION)` within the constructor?